ovs/lib/setup.py

```diff
@@ -905,19 +974,30 @@ def _setup_first_node(target_client, unique_id, cluster_name, node_name, hypervi
             logger.info('Setting up fleet')
             ServiceManager.setup_fleet()
 
-        print 'Setting up Arakoon'
-        logger.info('Setting up Arakoon')
-        result = ArakoonInstaller.create_cluster('ovsdb', cluster_ip, EtcdConfiguration.get('/ovs/framework/paths|ovsdb'), locked=False)
-        arakoon_ports = [result['client_port'], result['messaging_port']]
+        try:
+            arakooncompleted = EtcdConfiguration.get('/ovs/framework/hosts/{0}/arakooncompleted'.format(unique_id)) is True
+        except EtcdKeyNotFound:
+            arakooncompleted = None
+        if arakooncompleted is None:
+            print 'Setting up Arakoon'
```

Comment:
If the Etcd configuration value above exists, arakoon won't be configured, even if the Etcd configuration value was ```False``` or even ```None```. As in that case ```arakooncompleted``` will be ```False``` and this if-statement is false. If this is intended, you might rewrite this try/except and if by just using ```if EtcdConfiguration.exists(<key>):```